src/Utilities/Compiler/Extensions/CompilationExtensions.cs

```diff
@@ -26,5 +31,41 @@ internal static class CompilationExtensions
         /// <returns>The <see cref="INamedTypeSymbol"/> if found, null otherwise.</returns>
         internal static bool TryGetOrCreateTypeByMetadataName(this Compilation compilation, string fullTypeName, out INamedTypeSymbol namedTypeSymbol) =>
             WellKnownTypeProvider.GetOrCreate(compilation).TryGetOrCreateTypeByMetadataName(fullTypeName, out namedTypeSymbol);
+
+        /// <summary>
+        /// Gets a value indicating, whether the compilation of assembly targets .NET Framework.
+        /// This method differentiates between .NET Framework and other frameworks (.NET Core, .NET Standard, .NET 5 in future).
+        /// </summary>
+        /// <param name="compilation">The compilation</param>
+        /// <returns><c>True</c> if the compilation targets .NET Framework; otherwise <c>false</c>.</returns>
+        internal static bool DoesTargetDotNetFramework(this Compilation compilation)
+        {
+            // This condition should be removed in future.
+            // It only ensures that tests are successful, because tests compilate to .NET Framework.
+            if (compilation.ReferencedAssemblyNames.Any(identity => string.Equals(identity.Name, "netstandard", StringComparison.OrdinalIgnoreCase)))
+            {
+                return false;
+            }
```

Comment:
💭 I think we should remove this from the start. With a small amount of work to complete dotnet/roslyn-sdk#329 we should be able to get this working properly.